### PR TITLE
Fix windows closing

### PR DESCRIPTION
**Pull request by janisozaur**
I noticed an odd behaviour in calls to `Windows::Close()` and went through all cases, fixing where necessary. See individual commit messages. Please don't squash.

### CODE REVIEW

File: src/openrct2-ui/windows/EditorObjectSelection.cpp [hunk 1]
```diff
@@ -599,12 +599,14 @@ class EditorObjectSelectionWindow final : public Window
             if (!objectSelectResult.Successful)
                 return;
 
-            // Close any other open windows such as options/colour schemes to prevent a crash.
-            WindowCloseAll();
-            // WindowClose(*w);
+            auto& objRepository = OpenRCT2::GetContext()->GetObjectRepository();
```
Comment (duncanspumpkin) [inline src/openrct2-ui/windows/EditorObjectSelection.cpp:602]: have you checked the interaction with the options/colour schemes windows if they are open at this point?

File: src/openrct2-ui/windows/Guest.cpp [hunk 2]
```diff
@@ -207,7 +207,7 @@ class GuestWindow final : public Window
         {
             case WIDX_CLOSE:
                 Close();
-                break;
+                return;
```
[the remark this answers is not in the document]
Comment (duncanspumpkin) [reply src/openrct2-ui/windows/Guest.cpp:210]: This (the first switch) is the common widget input handler. After we have handled the SetPage the common widget has been handled and it doesn't need to propagate to the specific input handler as its a common widget.

[the remark this answers is not in the document]
Comment (janisozaur) [reply src/openrct2-ui/windows/Guest.cpp:210]: done

File: src/openrct2/platform/Crash.cpp [hunk 3]
```diff
@@ -56,7 +56,7 @@ static const wchar_t* _wszCommitSha1Short = WSZ("");
 static const wchar_t* _wszArchitecture = WSZ(OPENRCT2_ARCHITECTURE);
 static std::map<std::wstring, std::wstring> _uploadFiles;
 
-#    define BACKTRACE_TOKEN L"9d89dffbeb1c81deb554c848513c0fc57a3b91a35271ce44639b6748cf7b76f3"
+#    define BACKTRACE_TOKEN L"d3de8c689ea81ff21e030f1afeea5e293c0be6bfaae6975cd353c23fe6948322"
```
Comment (Broxzier) [inline src/openrct2/platform/Crash.cpp:59]: This seems unrelated to this PR

**Review by duncanspumpkin — APPROVED** [review]
